neofs/neofs_contract.go

```diff
@@ -568,6 +578,17 @@ func setConfig(ctx storage.Context, key, val interface{}) {
 	storage.Put(ctx, storageKey, val)
 }
 
+func hasNode(ctx storage.Context, key []byte, n common.IRNode) bool {
+	rawNode := std.Serialize(n)
+	it := storage.Find(ctx, key, storage.ValuesOnly)
+	for iterator.Next(it) {
+		if common.BytesEqual(rawNode, iterator.Value(it).([]byte)) {
+			return true
```

Comment:
Why `hasNode` compares serialized `common.IRNode{}` structures? It seems that now contract stores only public keys of those nodes. I suppose it should be 

```go
prefix := []byte(candidatesKey)
stKey := append(prefix, n.PublicKey...)
return storage.Get(ctx, stKey) != nil
```

Reply:
Fixed. My initial implementation was governed by the fact that `InnerRingCandidateAdd` can, in theory, update node info if the `IRNode` struct will be extended in future. Though old one compared public keys only so we are ok I thing.

Reply:
`hasNode` has no references now, remove?